### ISSUE TITLE
A defined key in a trait is not recognized by the classes that use the trait

### ISSUE THREAD

**Issue by ahmedvc** [umple/umple#1029]
Try the model below. Id will not be recognised as a key in the generated code.
trait API{
  id;
  key {id}
}

class Test{
  isA API;
}

**Comment by vahdat-ab** [comment]
@ahmedvc I hope you don't need this feature as an essential part of your work.
It's a little bit tricky because it's been implemented in that way for classes so it becomes hard to transfer it easily to traits. I need to refactor the code to get it working.

**Comment by ahmedvc** [comment]
No. It's not relevant to my work. I was testing traits in C++ when I found the issue. It is low priority for now.

**Comment by TimLethbridge** [comment]
We should deal with this at some point.
